lib/fuel-auth.js

```diff
@@ -20,19 +20,26 @@ module.exports = class FuelAuth {
 			if (typeof options.clientId !== 'string' || typeof options.clientSecret !== 'string') {
 				throw new Error('clientId or clientSecret must be strings');
 			}
+
+			if(options.apiVersion === 2 && !options.tenant){
+				throw new Error('Version 2 APIs require that you specify a tenant');
+			}
 		} else {
 			throw new Error('options are required. see readme.');
 		}
 
 		// set required values
 		this.accessToken = options.accessToken;
-		this.authUrl = options.authUrl || 'https://auth.exacttargetapis.com/v1/requestToken';
+		this.authUrl = options.authUrl || (options.apiVersion === 2 ? `https://${options.tenant}.auth.marketingcloudapis.com/v2/token` : 'https://auth.exacttargetapis.com/v1/requestToken');
```

Comment:
1. Let's rename the apiVersion to authVersion as this is specifically for OAuth2.
2. Constructing url will be done at the caller end. If you look at FuelSDK-Node repo, it sends the complete url. So caller end will send the complete url. So lets remove the tenant and url construction from tenant. just options.authUrl would do.

Reply:
Hi @manivinesh been off busy. Can you elaborate on your second point? Its unclear where the URL will be built otherwise? Unless we enforce that authUrl is required (which also doesnt sound stupid)